xml/src/test/java/org/ehcache/config/xml/XmlConfigurationTest.java

```diff
@@ -574,6 +577,26 @@ public void testDefaulSerializerXmlsSerializersValueHasWhitespaces() throws Exce
 
   }
 
+  @Test
+  public void testCacheEventNotificationServiceConfigurationTest() throws Exception {
+    final URL resource = XmlConfigurationTest.class.getResource("/configs/ehcache-cacheEventNotification.xml");
+    XmlConfiguration xmlConfig = new XmlConfiguration(resource);
+    assertThat(xmlConfig.getCacheConfigurations().size(), is(2));
+
+    Collection<?> configuration = xmlConfig.getCacheConfigurations().get("bar").getServiceConfigurations();
+    checkEventNotificationConfigurationExists(configuration);
+    Collection<?> templateConfiguration = xmlConfig.getCacheConfigurations().get("foo").getServiceConfigurations();
+    checkEventNotificationConfigurationExists(templateConfiguration);
```

Comment:
You can replace that whole logic / checks up to the `assertThat` by using `org.ehcache.spi.ServiceLocator#findSingletonAmongst`